### PR TITLE
Offsets refactor + add chest offset

### PR DESCRIPTION
**Pull request by Louka3000**
- Refactors the offsets to be linear in server, protocol and GUI.
- Fix some skeleton default rotation for hands
- Add a chest offset and remove the elbow offset

~~Protocol PR: https://github.com/SlimeVR/SolarXR-Protocol/pull/86~~ Merged

### CODE REVIEW

**Comment by Louka3000** [comment]
For GUI, I kinda like having the raw offsets, but that will require translation changes.
Most people won’t mess with manual body proportions, so trying to make it user friendly, even if it won’t ever be, is a loss. I would rather make it clear and precise.
What do you think @ImUrX @loucass003

**Comment by Eirenliel** [comment]
Please rebase

**Comment by Louka3000** [comment]
Done @Eirenliel

**Review by Eirenliel — APPROVED** [review]

**Comment by Eirenliel** [comment]
cannot webase in dithub.dew 🥺

**Review by ImUrX — CHANGES_REQUESTED** [review]
You shouldn't machine translate keys imo and should also add missing strings (no strict requirements on the last one but it should be done tbh)

**Comment by Louka3000** [comment]
I didn’t machine translate anything?

**Comment by Louka3000** [comment]
fixing missing keys tho

**Review by ImUrX — APPROVED** [review]

File: server/src/main/java/dev/slimevr/config/CurrentVRConfigConverter.java [hunk 1]
```diff
@@ -101,6 +101,41 @@ public ObjectNode convert(
...
+					float kneeHeight = offsetsNode.get("kneeHeight").floatValue();
+					offsetsNode.set("upperLegLength", new FloatNode(legsLength - kneeHeight));
+					offsetsNode.set("lowerLegLength", offsetsNode.get("kneeHeight"));
+					offsetsNode.remove("legsLength");
+					offsetsNode.remove("kneeHeight");
+
+					skeletonNode.set("offsets", offsetsNode);
+					modelData.set("skeleton", skeletonNode);
+				}
+			}
```
[the remark this answers is not in the document]
Comment (Louka3000) [reply server/src/main/java/dev/slimevr/config/CurrentVRConfigConverter.java:138]: absolutely

File: gui/src/components/onboarding/pages/body-proportions/autobone-steps/VerifyResults.tsx [hunk 2]
```diff
@@ -77,6 +77,7 @@ export function VerifyResultsStep({
                 </Typography>
               )}
             </div>
+            <div></div>
```
[the remark this answers is not in the document]
Comment (Louka3000) [reply gui/src/components/onboarding/pages/body-proportions/autobone-steps/VerifyResults.tsx:80]: I wanted to add some space and I hate HTML okay ;-;

[the remark this answers is not in the document]
Comment (Louka3000) [reply gui/src/components/onboarding/pages/body-proportions/autobone-steps/VerifyResults.tsx:80]: Okay I found out how to do it properly

File: server/src/main/java/dev/slimevr/vr/processor/skeleton/HumanSkeleton.java [hunk 3]
```diff
@@ -1655,81 +1663,98 @@ public void resetSkeletonConfig(SkeletonConfigOffsets config) {
...
-				height = vec.y;
+				float height = vec.y / BodyProportionError.eyeHeightToHeightRatio;
 				if (height > 0.5f) { // Reset only if floor level seems right,
 					skeletonConfig
 						.setOffset(
-							SkeletonConfigOffsets.TORSO,
-							((height) * 0.42f)
-								- skeletonConfig.getOffset(SkeletonConfigOffsets.NECK)
+							SkeletonConfigOffsets.CHEST,
+							(height * BodyProportionError.proportionLimits[2].getTargetRatio())
```
[the remark this answers is not in the document]
Comment (Louka3000) [reply server/src/main/java/dev/slimevr/vr/processor/skeleton/HumanSkeleton.java]: Done

**Review by ButterscotchV — APPROVED** [review]

**Review by Eirenliel — APPROVED** [review]

**Pull request merged**